client/dashboard/me/security-account-recovery/phone-number-input.tsx

```diff
@@ -0,0 +1,87 @@
+import { smsCountryCodesQuery } from '@automattic/api-queries';
+import { useQuery } from '@tanstack/react-query';
+import {
+	__experimentalHStack as HStack,
+	__experimentalInputControl as InputControl,
+	SelectControl,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { useEffect, useCallback } from 'react';
+import type { SecuritySMSNumber } from './types';
+
+import './style.scss';
+
+// TODO: We could potentially move this component to a shared place
+export default function PhoneNumberInput( {
+	data,
+	onChange,
+	isDisabled,
+}: {
+	data: SecuritySMSNumber;
+	onChange: ( value: Partial< SecuritySMSNumber > ) => void;
+	isDisabled: boolean;
+} ) {
+	const { data: smsCountryCodes, isLoading: isLoadingSmsCountryCodes } = useQuery(
+		smsCountryCodesQuery()
+	);
+
+	const countryCodes =
+		smsCountryCodes?.map( ( countryCode ) => ( {
+			label: countryCode.name,
+			value: countryCode.code,
+		} ) ) ?? [];
+
+	const onChangeCountryCode = useCallback(
+		( value: string ) => {
+			const countryCode = smsCountryCodes?.find( ( countryCode ) => countryCode.code === value );
+			if ( ! countryCode ) {
+				return;
+			}
+
+			return onChange( {
+				...data,
+				countryCode: countryCode.code,
+				countryNumericCode: countryCode.numeric_code,
+			} );
+		},
+		[ data, onChange, smsCountryCodes ]
+	);
+
+	// Set the initial country code if it's not already set
+	useEffect( () => {
```

Comment:
It's similar to the previous PR: we can preload the data, and then we don't need to use `useEffect()` here 😄